File: Agentic-Framework/function-tools/FunctionTool.java
package zepro;

import java.util.HashMap;
import java.util.Map;
import java.util.function.BiFunction;

public class FunctionTool {
	
	static BiFunction<Double, Double, Double> getDiscountedPrice  = (price, discount) ->
	       price - (price *discount / 100);
	       
	static class Agent {
		private Map<String, BiFunction<Double, Double, Double>> tools;
		public Agent(Map<String, BiFunction<Double, Double, Double>> tools) {
			this.tools = tools;
		}
		
		public String act(String task) {
            if (task.toLowerCase().contains("discount")) {
                double result = tools.get("getDiscountedPrice").apply(1000.0, 20.0);
                return "Final price after discount = " + result;
            }
            return "I don't know how to handle this yet.";
        }
	}
	
	public static void main(String[] args) {
        Map<String, BiFunction<Double, Double, Double>> tools = new HashMap<>();
        tools.put("getDiscountedPrice", getDiscountedPrice);

        Agent agent = new Agent(tools);
        System.out.println(agent.act("What is the price after 20% discount?"));
    }

}

File: Agentic-Framework/function-tools/FunctionTool.py

# A normal function
def get_discounted_process(price: float, discount: float) -> float:
    """Calculate discounted price"""
    return price - (price * discount / 100)


# Register function tools
tools = {
    "get_discounted_price": get_discounted_process
}

# Simulating an Agent that can call tools
class Agent:
    def __init__(self, tools: dict):
        self.tools = tools

    def act(self, task: str):
        # Very simple parser: looks for keywords in the task
        if "discount" in task.lower():
            tool = self.tools["get_discounted_price"]
            result = tool(1000, 20)  # hardcoded for example
            return f"Final price after discount = {result}"
        else:
            return "I don't know how to handle this yet."

# Run the agent
if __name__ == "__main__":
    agent = Agent(tools)
    response = agent.act("What is the price after 20% discount?")
    print(response)